src/main/java/com/bandwidth/sqs/publisher/MessagePublisher.java

```diff
@@ -0,0 +1,12 @@
+package com.bandwidth.sqs.publisher;
+
+import com.amazonaws.services.sqs.model.SendMessageResult;
+
+import java.time.Duration;
+import java.util.Optional;
+
+import io.reactivex.Single;
+
+public interface MessagePublisher<T> {
+    Single<SendMessageResult> publishMessage(T newMessage, String newQueueUrl, Optional<Duration> delay);
```

Comment:
Also, I already have the convenience method, almost exactly as you have it. (Except it uses empty instead of 0)